<launch>
    <arg name="kilotron" default="false" /> <!-- Kilotron or Gigatron? -->
	<arg name="map_file" default="$(find gigatron)/maps/nonexistent_map.yaml"/> <!-- placeholder for map file loading -->
	<arg name="map_server" default="false"/> <!-- launch map server or not -->
	<arg name="rviz_config_file" default="$(find gigatron)/rviz/odometry.rviz"/>

	<!-- run the map server with our map -->
	<node if="$(arg map_server)" name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />

	<!-- launch rviz -->
	<node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rviz_config_file)" output="screen" required="true" />

	<!-- publish odometry and tf based on wheel velocities -->
	<node name="odom_tf_publisher" type="odom_tf_publisher" pkg="gigatron">
		<param name="rate" value="200" />
		<param name="odom_topic" value="/odom" />
		<param name="state_topic" value="/state" />
		<rosparam file="$(find gigatron_hardware)/config/gigatron_platform.yaml" command="load" unless="$(arg kilotron)"/> 
    	<rosparam file="$(find gigatron_hardware)/config/kilotron_platform.yaml" command="load" if="$(arg kilotron)"/> 

	</node> 

</launch>